lib/delivery-locations-resolver.js

```
@@ -0,0 +1,117 @@
+        })
+        .catch((error) => {
+          // This is a common error:
+          //  "Error hitting SCSB API 502: <html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>502 Bad Gateway</h1></center>\r\n</body>\r\n</html>\r\n"
+          // return Promise.reject(error)
+          logger.info({ message: 'HTC API error. Send everything to NH', htcError: error.message })
+          return { [barcode]: 'NH' }
+        })
+    })
+  ).then((scsbResponses) => {
+    let ellapsed = ((new Date()) - __startAll)
+    logger.debug({ message: `HTC searchByParam API across ${barcodes.length} barcodes took ${ellapsed}ms total`, metric: 'searchByParam-barcode-multiple', timeMs: ellapsed })
+
+    // Filter out anything `undefined` and make sure at least one is valid:
+    var validPairs = [{}].concat(scsbResponses.filter((r) => r))
+    // Merge array of hashes into one big lookup hash:
+    return Object.assign.apply(null, validPairs)
+  })
+}
+
+function resolveDeliveryLocations (items) {
+  // Extract barcodes from items:
+  var barcodes = items.map((i) => i.identifier.filter((i) => /^urn:barcode:/.test(i))[0].split(':')[2])
+
+  // Get a map from barcodes to ReCAP customercodes:
+  return recapCustomerCodesByBarcodes(barcodes)
+    .then((barcodeToCustomerCode) => {
+      // Now map over items to affix deliveryLocations:
+      return items.map((item) => {
+        // Get this item's barcode:
+        var barcode = item.identifier.filter((i) => /^urn:barcode:/.test(i))[0].split(':')[2]
+
+        // If recap has a customer code for this barcode, map it by recap cust code:
+        if (barcodeToCustomerCode[barcode]) {
+          item.deliveryLocation = deliveryLocationsByCustomerCode(barcodeToCustomerCode[barcode])
+
+        // Otherwise, it's not in recap:
+        } else {
+          // Temporarily use deliveryLocation if it's indexed (deprecated) - otherwise populate by mapping:
+          if (!item.deliveryLocation) {
```

Yep. Now we're using the by-sierra-location mapping any time we have a holdingLocation to do the mapping. Still falling back on mocked data if we do not.